server/sonar-web/src/main/js/apps/code/utils.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import without from 'lodash/without';
+import sortBy from 'lodash/sortBy';
+
+import {
+    addComponent,
+    getComponent as getComponentFromBucket,
+    addComponentChildren,
+    getComponentChildren,
+    addComponentBreadcrumbs,
+    getComponentBreadcrumbs
+} from './bucket';
+import { getChildren, getComponent, getBreadcrumbs } from '../../api/components';
+import { translate } from '../../helpers/l10n';
+
+const METRICS = [
+  'ncloc',
+  'code_smells',
+  'bugs',
+  'vulnerabilities',
+  'duplicated_lines_density',
+  'alert_status'
+];
+
+const METRICS_WITH_COVERAGE = [
+  ...METRICS,
+  'coverage',
+  'it_coverage',
+  'overall_coverage'
+];
+
+const PAGE_SIZE = 100;
+
+function expandRootDir ({ components, total, ...other }) {
+  const rootDir = components.find(component => component.qualifier === 'DIR' && component.name === '/');
+  if (rootDir) {
+    return getChildren(rootDir.key, METRICS_WITH_COVERAGE).then(r => {
+      const nextComponents = without([...r.components, ...components], rootDir);
+      const nextTotal = total + r.components.length - /* root dir */ 1;
+      return { components: nextComponents, total: nextTotal, ...other };
+    });
+  } else {
+    return { components, total, ...other };
+  }
+}
+
+function prepareChildren (r) {
+  return {
+    components: r.components,
+    total: r.paging.total,
+    page: r.paging.pageIndex
+  };
+}
+
+function skipRootDir (breadcrumbs) {
+  return breadcrumbs.filter(component => {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Remove curly braces and "return" from this arrow function body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)